src/VisualStudio/Core/Def/Implementation/GenerateType/GenerateTypeDialogViewModel.cs

```diff
@@ -509,42 +509,45 @@ public int ProjectSelectIndex
         private Project _previouslyPopulatedProject = null;
         private List<DocumentSelectItem> _previouslyPopulatedDocumentList = null;
 
-        public IEnumerable<DocumentSelectItem> GetDocumentList(CancellationToken cancellationToken)
+        public IEnumerable<DocumentSelectItem> DocumentList
         {
-            if (_previouslyPopulatedProject == _selectedProject)
+            get
             {
-                return _previouslyPopulatedDocumentList;
-            }
+                if (_previouslyPopulatedProject == _selectedProject)
+                {
+                    return _previouslyPopulatedDocumentList;
+                }
 
-            _previouslyPopulatedProject = _selectedProject;
-            _previouslyPopulatedDocumentList = new List<DocumentSelectItem>();
+                _previouslyPopulatedProject = _selectedProject;
+                _previouslyPopulatedDocumentList = new List<DocumentSelectItem>();
 
-            // Check for the current project
-            if (_selectedProject == _document.Project)
-            {
-                // populate the current document
-                _previouslyPopulatedDocumentList.Add(new DocumentSelectItem(_document, "<Current File>"));
+                // Check for the current project
+                if (_selectedProject == _document.Project)
+                {
+                    // populate the current document
+                    _previouslyPopulatedDocumentList.Add(new DocumentSelectItem(_document, "<Current File>"));
 
-                // Set the initial selected Document
-                this.SelectedDocument = _document;
+                    // Set the initial selected Document
+                    this.SelectedDocument = _document;
 
-                // Populate the rest of the documents for the project
-                _previouslyPopulatedDocumentList.AddRange(_document.Project.Documents
-                    .Where(d => d != _document && !d.IsGeneratedCode(cancellationToken))
-                    .Select(d => new DocumentSelectItem(d)));
-            }
-            else
-            {
-                _previouslyPopulatedDocumentList.AddRange(_selectedProject.Documents
-                    .Where(d => !d.IsGeneratedCode(cancellationToken))
-                    .Select(d => new DocumentSelectItem(d)));
+                    // Populate the rest of the documents for the project
+                    _previouslyPopulatedDocumentList.AddRange(_document.Project.Documents
+                        .Where(d => d != _document && !d.IsGeneratedCode(CancellationToken.None))
```

Comment:
oh... this makes me so sad... is there no way we can support this in a non synchronous blocking manner?

Reply:
Probably. And given this was 100% broken for a year and a half...I'm not sure I care to find out.